docgen/assets/js/examples/tourism/index.js

```diff
@@ -0,0 +1,234 @@
+    if (pos.lat < bounds.se.lat) bounds.se.lat = pos.lat;
+
+    if (pos.lng < bounds.nw.lng) bounds.nw.lng = pos.lng;
+    if (pos.lng > bounds.se.lng) bounds.se.lng = pos.lng;
+    return bounds;
+  }, {
+    nw: {lat: -85, lng: 180},
+    se: {lat: 85, lng: -180},
+  });
+  const boundsConfig = fitBounds(boundingPoints, availableSpace);
+  const markers = hits.map(hit => <CustomMarker lat={hit.lat} lng={hit.lng} key={hit.objectID}></CustomMarker>);
+  const options = {
+    minZoomOverride: true,
+    minZoom: 1,
+  };
+  return (
+    <GoogleMap
+      options={() => options}
+      bootstrapURLKeys={{
+        key: 'AIzaSyAl60n7p07HYQK6lVilAe_ggwbBJFktNw8',
+      }}
+      center={boundsConfig.center}
+      zoom={boundsConfig.zoom}
+    >{markers}</GoogleMap>
+  );
+}
+
+HitsMap.propTypes = {
+  hits: PropTypes.object,
+};
+
+const ConnectedHitsMap = Hits.connect(HitsMap);
+
+class DatesAndGuest extends Component {
+  render() {
+    return (
+      <div className="row aisdemo-filter">
+        <div className="col-sm-2 aisdemo-filter-title">Dates</div>
+        <div className="col-sm-3"><input className="date form-control" value="10/30/3015" disabled /></div>
+        <div className="col-sm-3"><input className="date form-control" value="10/30/3015" disabled /></div>
+        <div className="col-sm-3"><div id="guests"></div></div>
+      </div>
+    );
+  }
+}
+
+const RoomType = RefinementList.connect(({items, refine, selectedItems}) => {
+  const itemComponents = items.map(item => {
+    const isSelected = selectedItems.indexOf(item.value) !== -1;
+    const value = isSelected ?
+      selectedItems.filter(v => v !== item.value) :
+      selectedItems.concat([item.value]);
+    const selectedClassName = isSelected ? ' ais-refinement-list--item__active' : '';
+    const itemClassName = `ais-refinement-list--item col-sm-3 ${selectedClassName}`;
+    return (
+      <div className={itemClassName} key={item.value}>
+        <div>
+          <label className="ais-refinement-list--label" onClick={e => {
+            e.preventDefault();
+            refine(value);
+          }}>
+            <input
+              type="checkbox"
+              className="ais-refinement-list--checkbox"
+              defaultChecked={isSelected ? 'checked' : ''}
+              />
+            {item.value}
+            <span className="ais-refinement-list--count">{item.count}</span>
+          </label>
+        </div>
+      </div>
+    );
+  });
+
+  return (
+    <div className="row aisdemo-filter">
+      <div className="col-sm-2 aisdemo-filter-title">Room Type</div>
+      <div id="room_types">
+        {itemComponents}
+      </div>
+    </div>
+  );
+});
+
+class Price extends Component {
+  render() {
+    return (
+      <div className="row aisdemo-filter">
+        <div className="col-sm-2 aisdemo-filter-title">Price Range</div>
+        <Range attributeName="price" />
+      </div>
+    );
+  }
+}
+
+const MyHits = Hits.connect(({hits}) => {
```

Comment:
Really like this syntax to explicit props. I will use it for the examples I wrote.
I was wondering if we shouldn't also use it in "Implementing your own xxx" js code sample section.